README.md

```diff
@@ -9,46 +9,66 @@
 
 </div>
 
+## Table of Contents
+
+-   [Overview](#overview)
+-   [Usage](#usage)
+    -   [General](#general)
+-   [Authorization](#authorization)
+-   [Setting the build name and build number when publishing build-info to Artifactory](#setting-the-build-name-and-build-number-when-publishing-build-info-to-artifactory)
+-   [Setting JFrog CLI version](#setting-jfrog-cli-version)
+-   [Downloading JFrog CLI from Artifactory](#downloading-jfrog-cli-from-artifactory)
+-   [Set up a FREE JFrog Environment in the Cloud](#set-up-a-free-jfrog-environment-in-the-cloud)
+-   [Example projects](#example-projects)
+-   [Contributions](#contributions)
+-   [References](#references)
+
+## Overview
+
 This GitHub Action downloads, installs and configures JFrog CLI, so that it can be used as part of the workflow.
 
 Additionally, the Action incorporates the following features when utilizing JFrog CLI to interact with the JFrog Platform:
-* Two distinct methods are available for authenticating with the JFrog Platform. Explore more details [here](#authorization)
-* There's no need to add the *build name* and *build number* options and arguments to commands which accept them.
-All build related operations will be automatically recorded with the *Workflow Name* as build name and *Run Number* as build number.
 
-# Usage
-## General
+-   Two distinct methods are available for authenticating with the JFrog Platform. Explore more details [here](#authorization)
+-   There's no need to add the _build name_ and _build number_ options and arguments to commands which accept them.
+    All build related operations will be automatically recorded with the _Workflow Name_ as build name and _Run Number_ as build number.
+
+## Usage
+
+### General
 
 ```yml
 - uses: jfrog/setup-jfrog-cli@v3
 - run: jf --version
 ```
-# Authorization
+
+## Authorization
+
 JFrog CLI operates in conjunction with the JFrog Platform. In order to facilitate this connection, certain connection details of the JFrog Platform must be provided.
 There exist two methods to provide these details, and you only need to choose **one** method:
-* The connection details of the JFrog platform can be stored as secrets. Read more about it [here](#storing-jfrog-connection-details-as-secrets).
-* The connection details of the JFrog platform can be auto-generated using OpenID Connect protocol. Read more about it [here](#connect-using-openid-connect).
 
+<details>
+    <summary>Storing JFrog connection details as secrets</summary>
```

Comment:
Storing JFrog connection details as secrets
-->
Connecting to JFrog using secrets